src/plots/xyPlot.ts

```diff
@@ -135,15 +237,28 @@ export class XYPlot<X, Y> extends Plot {
     var adjustCallback = key === XYPlot._X_KEY ? this._adjustYDomainOnChangeFromXCallback
                                                : this._adjustXDomainOnChangeFromYCallback;
     scale.onUpdate(adjustCallback);
+
+    if (this._lazyDomainChange) {
+      scale.offUpdate(this._renderCallback);
```

Comment:
The issue is that we need to not change the reference of the `this._renderCallback`. This means that we cannot plug and play a new `_rendererCallback`. This means that some sort of switch has to go inside the _rendererCallback. This means that the `Plot` needs to know about the fast version (since the `rendererCallback` is defined there)